### PR TITLE
EmittingPromise: Pass entire emitter object to resolver

### PR DESCRIPTION
**Pull request by allmarkedup**
@dkhuntrods so I think this works?? Test all pass and I can't think of any negative side affects. What do you think?

### CODE REVIEW

File: packages/internals/support/src/emitting-promise.js [hunk 1]
```diff
@@ -9,13 +9,13 @@ class EmittingPromise extends Promise {
...
     });
 
-    super((resolve, reject) => {
-      resolver(resolve, reject, (eventName, ...args) => {
-        process.nextTick(() => {
-          emitter.emit(eventName, ...args);
-        });
-      });
-    });
+    const oldEmit = emitter.emit;
```
Comment (dkhuntrods) [inline packages/internals/support/src/emitting-promise.js]: Could the function be bound once here, rather than in every call to `emit`?

**Comment by allmarkedup** [comment]
R.e. tests 'really' passing - yes that was a concern to me too but they are! Covering that is why I put in the 'counter' check for the number of `expects` around line 64. I'm sure an explicit `expects` counter for Chai probably exists as a plugin? Might be useful to integrate at some point.

**Comment by allmarkedup** [comment]
Oops didn't mean to close! :-)

File: packages/internals/support/src/emitting-promise.js (hunk 1 above)
Comment (allmarkedup) [inline packages/internals/support/src/emitting-promise.js]: Yes definitely! Good spot :-)

**Comment by allmarkedup** [comment]
@dkhuntrods and yes feel free to update and merge!

**Pull request merged**